presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixInitializer.java

```diff
@@ -43,74 +45,87 @@
  */
 public class RubixInitializer
 {
+    private static final RetryPolicy<?> RETRY_POLICY =
+            new RetryPolicy<>()
+                    // unlimited attempts
+                    .withMaxAttempts(-1)
+                    .withDelay(Duration.ofMillis(100))
+                    .withMaxDuration(Duration.ofMinutes(10));
+
     private static final Logger log = Logger.get(RubixInitializer.class);
 
+    private final RetryPolicy retryPolicy;
+    private final NodeManager nodeManager;
     private final CatalogName catalogName;
     private final RubixConfigurationInitializer rubixConfigurationInitializer;
     private final HdfsConfigurationInitializer hdfsConfigurationInitializer;
 
     @Inject
     public RubixInitializer(
+            NodeManager nodeManager,
+            CatalogName catalogName,
+            RubixConfigurationInitializer rubixConfigurationInitializer,
+            HdfsConfigurationInitializer hdfsConfigurationInitializer)
+    {
+        this(RETRY_POLICY, nodeManager, catalogName, rubixConfigurationInitializer, hdfsConfigurationInitializer);
+    }
+
+    @VisibleForTesting
+    RubixInitializer(
+            RetryPolicy retryPolicy,
+            NodeManager nodeManager,
             CatalogName catalogName,
             RubixConfigurationInitializer rubixConfigurationInitializer,
             HdfsConfigurationInitializer hdfsConfigurationInitializer)
     {
+        this.retryPolicy = retryPolicy;
+        this.nodeManager = nodeManager;
         this.catalogName = catalogName;
         this.rubixConfigurationInitializer = rubixConfigurationInitializer;
         this.hdfsConfigurationInitializer = hdfsConfigurationInitializer;
     }
 
-    public void initializeRubix(NodeManager nodeManager)
+    @PostConstruct
+    public void initializeRubix()
+    {
+        waitForCoordinator();
+        startRubix();
+    }
+
+    private void waitForCoordinator()
```

Comment:
How does the hive connector work until rubix is started?

The code here fails after 10 minutes. Does it mean the node is unusable at this point?
If so, it should ... shut down?
Or, maybe the caching is unusable, but the node is?

Reply:
When waiting for coordinator times-out then Presto node will fail to start. It will never announce itself as a worker